src/AutorestSwift/View Models/ClientMethodOptionsViewModel.swift

```diff
@@ -53,8 +53,25 @@ struct ClientMethodOptionsViewModel {
         }
         var properties = [PropertyViewModel]()
 
-        parameters.forEach {
-            properties.append(PropertyViewModel(from: $0.value))
+        // In generated code stencil, there is already a 'clientRequestId', so do not duplicate by adding anyone with same name
+        for parameter in parameters where parameter.value.name != "clientRequestId" {
+            properties.append(PropertyViewModel(from: parameter.value))
+        }
+
+        // For `Options` Group schema created by "x-ms-parameter-grouping" with postfix: Options,
+        // add all the properties from that group schema except property in path as the properties of the Method Option object
+        if let group = model.schemas.schema(for: groupName + name, withType: .group),
+            group.name.hasSuffix("Options") {
+            for property in group.properties ?? [] {
+                switch property {
+                case let .grouped(groupProperty):
+                    for param in groupProperty.originalParameter where param.paramLocation != .path {
+                        properties.append(PropertyViewModel(from: property.value))
+                    }
+                default:
+                    continue
+                }
+            }
```

Comment:
Is there an example of this? I think I get it, but the special casing concerns me somewhat.

Reply:
test swagger paging.json in an example.

In the swagger, it specifies 
```
            "name": "maxresults",
            "in": "header",
            "required": false,
            "type": "integer",
            "format": "int32",
            "description": "Sets the maximum number of items to return in the response.",
            "x-ms-parameter-grouping": {
                "postfix": "Options"
            }
          },
```

And this translates to code model:
```
         originalParameter:
            - !<!Parameter> &ref_33
              schema: *ref_2
              groupedBy: !<!Parameter> &ref_5
                schema: *ref_3
                implementation: Method
                language: !<!Languages> 
                  default:
                    name: pagingGetMultiplePagesOptions
                    description: Parameter group
                protocol: !<!Protocols> {}
              implementation: Method
              language: !<!Languages> 
                default:
                  name: maxresults
                  description: Sets the maximum number of items to return in the response.
                  serializedName: maxresults
              protocol: !<!Protocols> 
                http: !<!HttpParameter> 
                  in: header
```

Which Model4 build a `groupBy` parameter with `group name`+`operation name`Options (what specified in swagger as postfix).

Since we already have an Option class for each methods,   Autorest Swift just combine all properties into 1 Option object.

Reply:
Would it not make sense to just deconflict the names?

Reply:
technically, the names are not conflict, it just results in 2 `options` objects  with similar names. That is why I merge the 2 properties into 1 Options

![image](https://user-images.githubusercontent.com/8701731/100672673-23a1c200-3317-11eb-9001-ab6b01d3e647.png)

Reply:
And even if we don't merge the pagingGetMultiplePagesOptions into the existing opptions, we still need to add code to generate a model for 'pagingGetMultiplePagesOptions'.